core/src/impl/Kokkos_Default_GraphNode_Impl.hpp

```diff
@@ -35,8 +35,7 @@ namespace Impl {
 template <class ExecutionSpace>
 struct GraphNodeBackendSpecificDetails {
  private:
-  using execution_space_instance_storage_t =
-      ExecutionSpaceInstanceStorage<ExecutionSpace>;
+  using execution_space_instance_storage_t = InstanceStorage<ExecutionSpace>;
```

Comment:
I understand now that there was nothing enforcing it before but I think we should static assert that ExecutionSpace is indeed an execution space.
```suggestion
  static_assert(Kokkos::is_execution_space_v<ExecutionSpace>);
  using execution_space_instance_storage_t = InstanceStorage<ExecutionSpace>;
```